cmds/cmp/cmp.go

```diff
@@ -68,6 +69,20 @@ func openFile(name string) (*os.File, error) {
 	return f, err
 }
 
+// cmp is defined to fail with exit code 2
+func failf(format string, a ...interface{}) {
```

Comment:
why not log.Fatalf?

Reply:
Because of the comment above the function. According to the man-page: "Exit status is 0 if inputs are the same, 1 if different, 2 if trouble." The cases where this function is called are "trouble". Fatalf is defined to os.Exit(1) IIRC.

Reply:
But I agree that this is more a quick hack. Do we have such wrapper functions that return with a given error? Should we introduce them if not?

Reply:
I wonder where that comes from? 
Ah. here? 

EXIT STATUS
     The cmp utility exits with	one of the following values:

     0	   The files are identical.

     1	   The files are different; this includes the case where one file is
	   identical to	the first part of the other.  In the latter case, if
	   the -s option has not been specified, cmp writes to standard	error
	   that	EOF was	reached	in the shorter file (before any	differences
	   were	found).

     >1	   An error occurred.

bleah.

Reply:
wrappers and stuff would require more goroutine synchronization than what this command has right now. if you'd like to take it on, go for it? otherwise, we can merge as is.